File: packages/renderer-vue/playground/parts/CreatePartNode.ts
import { defineNode, NodeInterface } from "@baklavajs/core";
import { setType } from "@baklavajs/interface-types";
import { TextInputInterface, NumberInterface } from "../../src";
import { numberType, stringType, partType } from "../interfaceTypes";
import { Part } from "../models/Part";

export default defineNode({
    type: "CreatePartNode",
    title: "Create Part",
    inputs: {
        sku: () => new TextInputInterface("SKU", "").use(setType, stringType),
        description: () => new TextInputInterface("Description", "").use(setType, stringType),
        package: () => new TextInputInterface("Package", "").use(setType, stringType),
        quantity: () => new NumberInterface("Quantity", 0).use(setType, numberType),
    },
    outputs: {
        part: () => new NodeInterface<Part | undefined>("Part", undefined).use(setType, partType),
    },
    calculate(inputs) {
        return { part: new Part(inputs.sku, inputs.description, inputs.package, inputs.quantity) };
    },
});

File: BaklavaDependencyEngine/BaklavaDependencyEngine.Core/ExampleNodes/CreatePartNode.cs
using BaklavaDependencyEngine.Core.Models;

namespace BaklavaDependencyEngine.Core.ExampleNodes
{
    public class CreatePartNode : Node
    {
        public CreatePartNode() : base("CreatePartNode", "Create Part")
        {
        }

        protected override void InitializeInterfaces()
        {
            AddInput("sku", new NodeInterface("SKU", ""));
            AddInput("description", new NodeInterface("Description", ""));
            AddInput("package", new NodeInterface("Package", ""));
            AddInput("quantity", new NodeInterface("Quantity", 0));
            AddOutput("part", new NodeInterface("Part", null, false));
        }

        public override async Task<Dictionary<string, object>> Calculate(
            Dictionary<string, object> inputs,
            CalculationContext context)
        {
            var sku = inputs["sku"]?.ToString() ?? "";
            var description = inputs["description"]?.ToString() ?? "";
            var package = inputs["package"]?.ToString() ?? "";
            var quantity = Convert.ToInt32(inputs["quantity"]);

            var part = new Part(sku, description, package, quantity);

            return await Task.FromResult(new Dictionary<string, object>
            {
                ["part"] = part
            });
        }
    }
}